### DISCUSSION TITLE
Send MIDI when a stop is activated or deactivated

### DISCUSSION THREAD

**Discussion by newVideophile** [GrandOrgue/grandorgue#1749] [Help] [answered]
I am currently working on a modification of my second organ, and I would like to know if it is possible to ask GO to send the MIDI codes corresponding to the activated/deactivated stops when recalling a memorized registration. 

The reason is that I want to modify this organ to use it in headless mode. To replicate the stops status, I will use a LED strip with individually addressable LEDs. I will also use one capacitive sensor per stop to be able to select the stops. All this will be controlled by a Teensy, which will send the MIDI codes to GO. Handling individual stops will not be a problem, because the Teensy will know what capacitive sensor has been pressed, and thus which LED should be switch on or off accordingly. However, when recalling a registration that has been momorized in GO (I have also capacitive sensors acting as pistons), several stops are simultaneously activated, and there is no way for the Teensy to know which ones. Unless GO can send all the corresponding MIDI codes, as if the stops had been manually activated one by one.

Of course, I could use the Teensy for the registration memory management (this is what I am doing with my main organ), but I see this as a plan B only.

**Comment by oleg68** [comment, marked as answer]
GO can send midi events for stops when they are activated and deactivated.

Just right-click on the stop button, go to the `Send` tab and assign a midi event to it.

**Reply by newVideophile** [reply]
Thank you oleg !

**Comment by larspalo** [comment]
It should be possible. Of course, depending on your actual MIDI hardware... Use the Send tab of the normal MIDI dialog that pops up when you right click an element.

On one setup of GO that I regularly use I have a [AKAI Professional APC mini MK2](https://www.thomann.de/se/akai_professional_apc_mini_mk2.htm) setup for various controls (Stops, Setter buttons, Temperament changes etc). After I've programmed the pad lights to show up on activation, whenever that stop (or other button) is active it will light up on the device.

**Reply by newVideophile** [reply]
My MIDI hardware will be a Teensy, a LED strip with 60 LEDs (more than enough for Friesach) and DIY capacitive sensors

**Reply by larspalo** [reply]
@newVideophile If it's not too much of trouble for you or breaking any laws, please document the whole process both with text and images and make a post about it. It might inspire others!

**Reply by newVideophile** [reply]
Sure! It is a fully personal project, I don't see any possibility to break any law here.

My two organs are shown here: https://github.com/GrandOrgue/grandorgue/discussions/875#discussioncomment-3300854

The one I will modify is on the last picture. My end goal is to remove the laptop and use a headless Raspberry Pi instead. 

Be patient, though, because I am preparing a wedding in March, so I have to devote my free time to this in priority. I will start during the Christmas holidays (I have all the components already), but the is no ETA yet.

**Reply by larspalo** [reply]
@newVideophile That's great! Include as much details as possible. Always think of what someone else, without your previous knowledge and experience, would understand from it and be able to do with that information!

As I understand it, the latest generation of Pi will be quite capable and could easily power a reasonably sized organ in GO without any problems. At least I think your endeavours with this project would be very interesting to follow!

**Comment by newVideophile** [comment]
> @newVideophile That's great! Include as much details as possible. Always think of what someone else, without your previous knowledge and experience, would understand from it and be able to do with that information!

I'll do my best!

> As I understand it, the latest generation of Pi will be quite capable and could easily power a reasonably sized organ in GO without any problems. At least I think your endeavours with this project would be very interesting to follow!

The Rapsberry Pi 4 with 8 Gb of RAM can handle Friesach, if you accept a few concessions. I use it on my main organ already. The new Pi 5 brings more CPU but the same amount of RAM. Unfortunately, the opposite would have been better...
